### ISSUE TITLE
Error when run plan-all with cross-references between modules

### ISSUE THREAD

**Issue by mazzy89** [gruntwork-io/terragrunt#435]
One of the most powerful (imho) feature of Terragrunt is to define dependencies between modules. Let's assume we have the following scenario:

```
.
|-- backend
|   `-- terraform.tfvars
|-- ssh-keypair
|   `-- terraform.tfvars
`-- vpc
    `-- terraform.tfvars
```

The module `backend` use the ssh keypair resource created in the module `ssh-keypair`. Inside the module `backend` I have something like this:

```hcl
data "terraform_remote_state" "ssh_keypair" {
  backend = "s3"

  config {
    bucket = "${var.ssh_keypair_remote_tfstate_bucket}"
    key = "${var.ssh_keypair_remote_tfstate_key}"
    region = "${var.aws_region}"
  }
}

module "autoscaling" {

  key_name  = "${data.terraform_remote_state.ssh_keypair.ssh_key_name}"

}
```

All good you'd say. However there is a drawback to consider. In the above scenario a `plan-all` doesn't work because when the backend module runs, it fails with the error that the `ssh_key_name` reference doesn't exist. In order to fix it, you need first `apply` the `ssh-keypair` module and only after you can execute a plan.

How to mitigate as much as possible this issue? Any workaround? Any thoughts?

I was thinking at least to add in the README a warning that `plan-all` fails when there is a cross reference between module

**Comment by brikis98** [comment]
Agreed that it's not idea, but there isn't really a workaround. If you have run `apply` in a module, and you depend on that module, then the `plan` _should_ fail, as your dependencies are, in fact, missing! Closing in favor of https://github.com/gruntwork-io/terragrunt/issues/262 which has a discussion of possible alternatives.

**Comment by mazzy89** [comment]
> If you have run apply in a module, and you depend on that module, then the plan should fail, as your dependencies are, in fact, missing! 

That make perfectly sense. Thank you for the fast reply and reaction.
